R/utils-read.R

```diff
@@ -626,15 +626,15 @@ readEstimatedTimeSeries <- function(reportObject, seriesName, timezone, descript
         p <- estimatedPeriods[i,]
         startTime <- p$start
         endTime <- p$end
-        if(inverted){
-          estimatedSubset <- rbind(estimatedSubset, seriesData[['points']][seriesData[['points']][['time']] < startTime | seriesData[['points']][['time']] >= endTime,])
-        } else {
-          estimatedSubset <- rbind(estimatedSubset, seriesData[['points']][seriesData[['points']][['time']] >= startTime & seriesData[['points']][['time']] < endTime,])
-        }
+        estimatedSubset <- rbind(estimatedSubset, seriesData[['points']][seriesData[['points']][['time']] >= startTime & seriesData[['points']][['time']] < endTime,])
       }
     }
     #Replace data with only saved data
-    seriesData[['points']] <- estimatedSubset
+    if(inverted){
+      seriesData[['points']] <- seriesData[['points']][ !(seriesData[['points']][['time']] %in% estimatedSubset[['time']]), ]
```

Comment:
I think you can just use `subset` to make it easier. It's what we use in `subsetByMonth`. Here's my reproducible example using test data:

```r
library(jsonlite)

reportObject <- fromJSON(system.file('extdata','testsnippets','test-time-series.json', package = 'repgen'))
tz <- repgen:::fetchReportMetadataField(reportObject, "timezone")
series <- repgen:::readTimeSeries(reportObject, "testSeries1", repgen:::fetchReportMetadataField(reportObject, "timezone"))

startE <- repgen:::flexibleTimeParse(series[['estimatedPeriods']][['startDate']], tz)
endE <- repgen:::flexibleTimeParse(series[['estimatedPeriods']][['endDate']], tz)


estTimeSeries <- subset(series[['points']], time >= startE & time <= endE)
nonEstTimeSeries <- subset(series[['points']], time < startE | time > endE)
```

I think your code could change to (except I don't really follow the circumstance surrounding the inverted scenario):
```r
for(i in 1:nrow(estimatedPeriods)) {
    p <- estimatedPeriods[i,]
    startTime <- p$start
    endTime <- p$end
    if(inverted){
      estimatedSubset_i <- subset(estimatedSubset, time >= startTime & time < endTime)
    } else {
      estimatedSubset_i <- ?
    }
    estimatedSubset <- rbind(estimatedSubset, estimatedSubset_i)
}
```